# Description: Compile in the bits to allow Oxide to provide its own
#  WebContentsView implementation
# Author: Chris Coulson <chris.coulson@canonical.com>

diff --git a/content/content_browser.gypi b/content/content_browser.gypi
--- a/content/content_browser.gypi
+++ b/content/content_browser.gypi
@@ -1419,24 +1419,24 @@
       'browser/web_contents/aura/window_slider.h',
       'browser/web_contents/touch_editable_impl_aura.cc',
       'browser/web_contents/touch_editable_impl_aura.h',
       'browser/web_contents/web_contents_android.cc',
       'browser/web_contents/web_contents_android.h',
       'browser/web_contents/web_contents_impl.cc',
       'browser/web_contents/web_contents_impl.h',
       'browser/web_contents/web_contents_view.h',
-      'browser/web_contents/web_contents_view_android.cc',
-      'browser/web_contents/web_contents_view_android.h',
-      'browser/web_contents/web_contents_view_aura.cc',
-      'browser/web_contents/web_contents_view_aura.h',
+      #'browser/web_contents/web_contents_view_android.cc',
+      #'browser/web_contents/web_contents_view_android.h',
+      #'browser/web_contents/web_contents_view_aura.cc',
+      #'browser/web_contents/web_contents_view_aura.h',
       'browser/web_contents/web_contents_view_guest.cc',
       'browser/web_contents/web_contents_view_guest.h',
-      'browser/web_contents/web_contents_view_mac.h',
-      'browser/web_contents/web_contents_view_mac.mm',
+      #'browser/web_contents/web_contents_view_mac.h',
+      #'browser/web_contents/web_contents_view_mac.mm',
       'browser/web_contents/web_drag_dest_mac.h',
       'browser/web_contents/web_drag_dest_mac.mm',
       'browser/web_contents/web_drag_source_mac.h',
       'browser/web_contents/web_drag_source_mac.mm',
       'browser/web_contents/web_drag_utils_win.cc',
       'browser/web_contents/web_drag_utils_win.h',
       'browser/webui/content_web_ui_controller_factory.cc',
       'browser/webui/content_web_ui_controller_factory.h',
@@ -1459,16 +1459,18 @@
       'browser/webui/web_ui_message_handler.cc',
       'browser/zygote_host/zygote_host_impl_linux.cc',
       'browser/zygote_host/zygote_host_impl_linux.h',
       'zygote/zygote_linux.cc',
       'zygote/zygote_linux.h',
       'zygote/zygote_main_linux.cc',
       '<(DEPTH)/../../../shared/port/content/browser/power_save_blocker_oxide.cc',
       '<(DEPTH)/../../../shared/port/content/browser/power_save_blocker_oxide.h',
+      '<(DEPTH)/../../../shared/port/content/browser/web_contents_view_oxide.cc',
+      '<(DEPTH)/../../../shared/port/content/browser/web_contents_view_oxide.h',
     ],
     'android_browser_sources': [
       'browser/android/java/gin_java_bound_object.cc',
       'browser/android/java/gin_java_bound_object.h',
       'browser/android/java/gin_java_bound_object_delegate.cc',
       'browser/android/java/gin_java_bound_object_delegate.h',
       'browser/android/java/gin_java_bridge_dispatcher_host.cc',
       'browser/android/java/gin_java_bridge_dispatcher_host.h',
